docs/book/src/migration/guide.md

```diff
@@ -0,0 +1,192 @@
+# Migration Guide
+
+This document describes how to migrate a project from kubebuilder v1 to v2.
+
+We suggest you read [this doc](./v1vsv2.md) that compares the notable differences between v1 and v2.
+
+The recommended way of migration is to create a new v2 project and copy the business logic code.
+
+Please ensure you have followed the [installation guide](../quick-start.md#installation) to install the required
+components.
+
+We have an [example v1 project](https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/migration/testdata/gopath/project). It should look like the
+[v2 project in the cronjob tutorial](https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/cronjob-tutorial/testdata/project) after the migration.
+
+## Preparation
+
+First, we need to figure out what are the domain, group, version and kind for kubebuilder.
+
+In your kubebuilder v1 project, the `pkg` directory should look like this.
+
+```
+pkg/
+├── apis
+│   ├── addtoscheme_{your-group}_{your-version}.go
+│   ├── apis.go
+│   └── {your-group}
+│       ├── group.go
+│       └── {your-version}
+│           ├── {your-kind-in-lower-case}_types.go
+│           ├── {your-kind-in-lower-case}_types_test.go
+│           ├── doc.go
+│           ├── register.go
+│           ├── {your-version}_suite_test.go
+│           └── zz_generated.deepcopy.go
+├── controller
+└── webhook
+```
+
+In `register.go`, you can find something like
+```go
+SchemeGroupVersion = schema.GroupVersion{Group: {your-group}.{your-domain}, Version: {your-version}}
+```
+
+Now you have figured out what are your domain, group, version and kind.
+
+## Initialize the Project
+
+Run the following command
+
+```bash
+kubebuilder init --domain {your-domain}
+```
+
+## Migrate API and Controller
+
+Run the following command to scaffold the API and the controller. Please follow the prompt to scaffold the API and (or)
```

Comment:
ditto here for the conversational tone and specific example:

> Next, we'll re-scaffold out the controller and API types.  SInce we want both, we'll say yes to both the API and controller prompts when asked what parts we want to scaffold: